### DISCUSSION TITLE
How to run `vim.ui.select` synchronously?

### DISCUSSION THREAD

**Discussion by FoamScience** [stevearc/dressing.nvim#70] [Q&A] [answered]
The idea is to use it in configuration files

Something along these lines:
```lua
dap.configurations.cpp = {
 {
    program = function()
        local var = ""
        vim.ui.select({'main', 'bin'}, {prompt = 'Choose binary:'}, function(choice) var = choice end)
        return var
    end,
...}
```
But by the time the configuration is read, it appears `vim.ui.select` is executed asynchronously (and program is nil) but I need the function to block
and wait for the selection.

I'm new to the whole Lua thing, so I thought I would ask if there is a simple way to do this?

**Comment by FoamScience** [comment, marked as answer]
Found a piece of code demonstrating this [here in  `dap.ui`](https://github.com/mfussenegger/nvim-dap/blob/66d33b7585b42b7eac20559f1551524287ded353/lua/dap/ui.lua#L55)

**Reply by rohit-kumar-j** [reply]
After searching for soooo.. long 

Thanks @FoamScience!! 

Here is what I am doing with this: [Multit Terminal Support with CMakeTools](https://github.com/Civitasv/cmake-tools.nvim/pull/58)
